lightning/src/offers/invoice_request.rs

```diff
@@ -2981,4 +3010,31 @@ mod tests {
 			Err(_) => panic!("unexpected error"),
 		}
 	}
+
+	#[test]
+	fn test_string_truncate_safe() {
+		// We'll correctly truncate to the nearest UTF-8 code point boundary:
+		// ❤      variation-selector
+		// e29da4 efb88f
+		let s = String::from("❤️");
+		for idx in 0..(s.len() + 5) {
```

Comment:
I had to think whether the 5 comes from somewhere. I think it is just 'go somewhat past len'?